### PR TITLE
Implement Speed Detection

### PR DESCRIPTION
**Pull request by vectrixdevelops**
This is nearly ready to go to be merged.

Adds in a lot of fixes to "everything" and adds in a speed detection to kick off testing for future features such as the punishment system. This should make the project in a state where people can start making their own detections.

### CODE REVIEW

File: src/main/java/io/github/connorhartley/guardian/sequence/condition/ConditionResult.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+package io.github.connorhartley.guardian.sequence.condition;
+
+import io.github.connorhartley.guardian.sequence.report.SequenceReport;
+
+public class ConditionResult {
+
+    private final boolean pass;
+    private final SequenceReport sequenceReport;
+
+
```
Comment (Meronat) [inline src/main/java/io/github/connorhartley/guardian/sequence/condition/ConditionResult.java]: Extra line

File: src/main/java/io/github/connorhartley/guardian/util/StorageValue.java [hunk 2]
```diff
@@ -24,181 +24,189 @@
...
-                else
-                    return Optional.of((V) node.getValue(new TypeToken(this.defaultValue.getClass()){}, this.defaultValue));
-            } catch(Exception e) {
-                return Optional.of(this.defaultValue);
+                if (this.valueTypeToken != null) {
+                    return node.getValue(this.valueTypeToken, this.defaultValue);
+                } else {
+                    return node.getValue(new TypeToken<E>(this.defaultValue.getClass()) {}, this.defaultValue);
+                }
+            } catch (Exception e) {
```
Comment (Meronat) [inline src/main/java/io/github/connorhartley/guardian/util/StorageValue.java]: Is this an acceptable case to occur? Should something be logged warning the owner this has happened?

File: src/main/java/io/github/connorhartley/guardian/sequence/SequenceListener.java [hunk 3]
```diff
@@ -23,27 +23,31 @@
...
-                new IllegalStateException("SequenceListener was attempted to be loaded before UserStorageService object was provided."));
     }
 
     /*
      * Listeners will go here. For example:
      *
      * @Listener
-     * public void onEntityCollide(CollideEntityEvent event, @First Player player) {
-     *     this.userStorageService.get(player.getUniqueId()).ifPresent(user -> this.sequenceController.invoke(user, event));
+     * public void onEntityCollide(CollideEntityEvent event, @Getter("getTargetEntity") Player player) {
```
Comment (Meronat) [inline src/main/java/io/github/connorhartley/guardian/sequence/SequenceListener.java]: Why is this example needed when the exact code is essentially right below?

**Comment by parlough** [comment]
So much excitement.

**Comment by parlough** [comment]
Close this elderly PR.

**Comment by parlough** [comment]
👵⭕️👴

**Pull request closed without merging**